packages/components/pie-modal/src/defs.ts

```diff
@@ -14,6 +14,19 @@ export interface ModalProps {
      * When true, the modal will be open.
      */
     isOpen: boolean;
+    /**
+     * When set to `true`:
+     *  1. The close button within the modal will be visible.
+     *  2. The user can dismiss the modal via the ESCAPE key, clicking the backdrop
+     *     or via a close button.
+     *
+     * When set to `false`:
+     *  1. The close button within the modal will be hidden.
+     *  2. The user can NOT dismiss the modal via the ESCAPE key or clicking the backdrop.
+     *
+     */
+    isDismissible: boolean;
```

Comment:
I have been wondering about our usage of boolean properties and whether they would be more useful as attributes. [Spectrum takes this approach](https://opensource.adobe.com/spectrum-web-components/components/dialog/#dismissable) with their modal/components too.

If the attribute is present, that's treated as true. If the attribute is absent, that's treated as false.

@kevinrodrigues @ashleynolan @xander-marjoram I'm keen to hear what you all think?

Reply:
All we'd need to do is set `Reflect: true` on the property here, I believe

Reply:
Fixed.

Reply:
Sorry I'm late to this! I'm personally of the opinion it's much clearer when Boolean props are named with `is`/`has` on them.

If we were to go down this route in the future of changing this approach, we'd want to do it everywhere, rather than just on a single instance here, otherwise we'll end up with inconsistencies.

Would it be a pain to change this back for now and we can discuss how we want to name these separately outside of this PR?

Reply:
I agree. Our components will undoubtedly get more complex as they grow so I think having a consistent group of props that signify a particular given value makes sense. Will change back for now.

Reply:
I do understand the sentiment but I wonder if this pattern is still as useful when we're using typescript? 

In terms of external consumers, if they're adding a prop to a component then they're aware of what that prop is. So I'm struggling to decide if this is still as useful as it was when we were using vanilla JS in our Vue library/Legacy JS packages.

Happy to leave it for now of course, but would be great to chat further!